phpBB/phpbb/db/extractor/mysql_extractor.php

```diff
@@ -0,0 +1,362 @@
+<?php
+/**
+*
+* This file is part of the phpBB Forum Software package.
+*
+* @copyright (c) phpBB Limited <https://www.phpbb.com>
+* @license GNU General Public License, version 2 (GPL-2.0)
+*
+* For full copyright and license information, please see
+* the docs/CREDITS.txt file.
+*
+*/
+
+namespace phpbb\db\extractor;
+
+class mysql_extractor extends base_extractor
+{
+	/**
+	* {@inheritdoc}
+	*/
+	public function write_start($table_prefix)
+	{
+		$sql_data = "#\n";
+		$sql_data .= "# phpBB Backup Script\n";
+		$sql_data .= "# Dump of tables for $table_prefix\n";
+		$sql_data .= "# DATE : " . gmdate("d-m-Y H:i:s", $this->time) . " GMT\n";
+		$sql_data .= "#\n";
+		$this->flush($sql_data);
+	}
+
+	/**
+	* {@inheritdoc}
+	*/
+	public function write_table($table_name)
+	{
+		static $new_extract;
+
+		if ($new_extract === null)
+		{
+			if ($this->db->get_sql_layer() === 'mysqli' || version_compare($this->db->sql_server_info(true), '3.23.20', '>='))
+			{
+				$new_extract = true;
+			}
+			else
+			{
+				$new_extract = false;
+			}
+		}
+
+		if ($new_extract)
+		{
+			$this->new_write_table($table_name);
+		}
+		else
+		{
+			$this->old_write_table($table_name);
+		}
+	}
+
+	/**
+	* {@inheritdoc}
+	*/
+	public function write_data($table_name)
+	{
+		if ($this->db->get_sql_layer() === 'mysqli')
+		{
+			$this->write_data_mysqli($table_name);
+		}
+		else
+		{
+			$this->write_data_mysql($table_name);
+		}
+	}
+
+	/**
+	* Extracts data from database table (for MySQLi driver)
+	*
+	* @param	string	$table_name	name of the database table
+	* @return null
+	*/
+	private function write_data_mysqli($table_name)
```

Comment:
@nickvergessen it is possible, but those methods are ugly by default, for example using mysql_ and mysqli_ functions